### PR TITLE
Backend: TimeLimitedCache

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Changes TimeLimitedCache to extend MutableMap instead of Iterator. Also removed some deprecated functions, and deprecated some other ones. Also removed the deprecation on .cancel() inside LorenzEvent

## Changelog Technical Details
+ Changed `TimeLimitedCache` to extend `MutableMap` instead of `Iterator`. - Empa

### CODE REVIEW

**Comment by hannibal002** [comment]
> Also removed the deprecation on .cancel() inside LorenzEvent

since the whole of lorenz event is deprecated, there is no need to use the function. this is just an additional marker that when using deprecated functions you want to migrate them over to skyhanni event, like every other instance of lorenz event as well

**Comment by ItsEmpa** [comment]
> > Also removed the deprecation on .cancel() inside LorenzEvent
> 
> since the whole of lorenz event is deprecated, there is no need to use the function. this is just an additional marker that when using deprecated functions you want to migrate them over to skyhanni event, like every other instance of lorenz event as well

i feel like you should only actually be able to see that LorenzEvent is deprecated if you want to create a new event. if you want to cancel an already existing event (such as cancelling a message with LorenzChatEvent), you shouldnt be told that the function to cancel the event is deprecated.

File: src/main/java/at/hannibal2/skyhanni/utils/TimeLimitedSet.kt [hunk 1]
```diff
@@ -26,13 +27,15 @@ class TimeLimitedSet<T : Any>(
...
+    operator fun minusAssign(element: T) {
+        remove(element)
+    }
 
     operator fun contains(element: T): Boolean = cache.containsKey(element)
 
     fun clear() = cache.clear()
 
-    fun toSet(): Set<T> = HashSet(cache.keys())
+    fun toSet(): MutableSet<T> = cache.keys
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/TimeLimitedSet.kt]: im not sure i like the idea of exposing mutable cache outside this class. this allows unintended behaviour

**Review by hannibal002 — CHANGES_REQUESTED** [review]
lets discuss

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt [hunk 2]
```diff
@@ -51,7 +51,7 @@ object FishingTimer {
 
     private var mobsToFind = 0
 
-    private val recentMobs = TimeLimitedSet<Mob>(2.seconds)
+    private var recentMobs = TimeLimitedSet<Mob>(2.seconds)
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: Is there a reason to make those var?

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: After my commit the `object += value` requires this for the add function. Not sure if this is the correct solution though

Comment (ItsEmpa) [reply src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt]: reverted the changes in that commit that hanni made

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**